src/rotate.js

```diff
@@ -0,0 +1,87 @@
+function isNumeric (number) {
+  return !isNaN(number);
+}
+
+function InvalidArgument (message) {
+  this.message = message;
+  this.name = 'InvalidArgument';
+}
+
+function parse (input) {
+  if (input === undefined || input === 'none') {
+    return null;
+  }
+
+  var values = input.split(' ');
+  var numValues = values.length;
+
+  if (numValues !== 4 && numValues !== 1) {
+    throw new InvalidArgument('Incorrect number of arguments for rotate');
+  }
+
+  var angle = values[numValues - 1];
+  var angleUnitArray = /(deg|grad|rad|turn)$/.exec(angle);
+
+  if (angleUnitArray === null) {
+    throw new InvalidArgument('Angle units must be one of: deg, grad, rad or turn');
+  }
+
+  var unit = angleUnitArray[0];
+  var number = angle.substring(0, angle.length - unit.length);
+  if (!(isNumeric(number))) {
+    throw new InvalidArgument('Angle given must be a number followed by units');
+  }
+
+  if (numValues > 1) {
+    for (var i = 0; i < numValues - 1; i++) {
+      if (!(isNumeric(values[i]))) {
+        throw new InvalidArgument('Axis value must be a number');
+      }
+    }
+  }
+
+  // numValues - 1 --> because the last values of the array contains unit characters
+  for (var i = 0; i < numValues - 1; i++) {
+    values[i] = Number(values[i]);
+  }
+  values[i] = Number(number);
```

Comment:
values[numValues - 1]  is clearer than  values[i]

Also, should we convert to degrees?